src/gui/sdlmain.cpp

```diff
@@ -2222,11 +2219,14 @@ int main(int argc, char* argv[]) {
 	if (SDL_Init_Wrapper() < 0)
 		E_Exit("Can't init SDL %s", SDL_GetError());
 	sdl.inited = true;
+	// Once initialized, ensure we clean up SDL for all exit conditions
+	atexit(SDL_Quit);
```

Comment:
This pattern of registering the `Quit` handler immediately after initializing SDL is from the SDL2 docs - https://wiki.libsdl.org/SDL_Quit.